syntax = "proto3";

package com.centreon.broker;

message Service {
  uint64 host_id = 1;
  uint64 service_id = 2;

  enum AckType {
    ACKNOWLEDGEMENT_NONE = 0;
    ACKNOWLEDGEMENT_NORMAL = 1;
    ACKNOWLEDGEMENT_STICKY = 2;
  }
  bool acknowledged = 3;
  AckType acknowledgement_type = 4;

  bool active_checks_enabled = 5;
  bool enabled = 6;
  int32 downtime_depth = 7;
  string check_command = 8;
  int32 check_interval = 9;
  string check_period = 10;

  enum CheckType {
    CHECK_ACTIVE = 0;
    CHECK_PASSIVE = 1;
  }
  CheckType check_type = 11;
  int32 current_check_attempt = 12;
  enum State {
    OK = 0;
    WARNING = 1;
    CRITICAL = 2;
    UNKNOWN = 3;
    PENDING = 4;
  }
  State current_state = 13;
  bool event_handler_enabled = 14;
  string event_handler = 15;
  double execution_time = 16;
  bool flap_detection_enabled = 17;
  bool has_been_checked = 18;
  bool is_flapping = 19;
  int64 last_check = 20;
  State last_hard_state = 21;
  int64 last_hard_state_change = 22;
  int64 last_notification = 23;
  int32 notification_number = 24;
  int64 last_state_change = 25;
  int64 last_time_ok = 26;
  int64 last_time_warning = 27;
  int64 last_time_critical = 28;
  int64 last_time_unknown = 29;
  int64 last_update = 30;
  double latency = 31;
  int32 max_check_attempts = 32;
  int64 next_check = 33;
  int64 next_notification = 34;
  bool no_more_notifications = 35;
  bool notifications_enabled = 36;
  string output = 37;
  string long_output = 38;
  bool passive_checks_enabled = 39;
  double percent_state_change = 40;
  string perf_data = 41;
  double retry_interval = 42;
  string host_name = 43;
  string service_description = 44;
  bool should_be_scheduled = 45;
  bool obsess_over = 46;

  enum StateType {
    SOFT = 0;
    HARD = 1;
  }

  StateType state_type = 47;
}
